src/gpodder/sync.py

```diff
@@ -307,214 +303,124 @@ def get_free_space(self):
-                self.remove_track_gpod(track)
-
     def get_all_tracks(self):
         tracks = []
-        for track in gpod.sw_get_playlist_tracks(self.podcasts_playlist):
-            filename = gpod.itdb_filename_on_ipod(track)
+        for track in self.ipod.get_podcast_tracks():
+            filename = track.filename_on_ipod
 
             if filename is None:
-                # This can happen if the episode is deleted on the device
-                logger.info('Episode has no file: %s', track.title)
-                self.remove_track_gpod(track)
-                continue
-
-            length = util.calculate_size(filename)
-            timestamp = util.file_modification_timestamp(filename)
-            modified = util.format_date(timestamp)
-            try:
-                released = gpod.itdb_time_mac_to_host(track.time_released)
-                released = util.format_date(released)
-            except ValueError as ve:
-                # timestamp out of range for platform time_t (bug 418)
-                logger.info('Cannot convert track time: %s', ve)
-                released = 0
-
-            t = SyncTrack(track.title, length, modified,
-                    modified_sort=timestamp,
-                    libgpodtrack=track,
+                length = 0
+                modified = ''
+            else:
+                length = util.calculate_size(filename)
+                timestamp = util.file_modification_timestamp(filename)
+                modified = util.format_date(timestamp)
+
+            t = SyncTrack(track.episode_title, length, modified,
+                    ipod_track=track,
                     playcount=track.playcount,
-                    released=released,
-                    podcast=track.artist)
+                    podcast=track.podcast_title)
             tracks.append(t)
         return tracks
 
+    def episode_on_device(self, episode):
+        return next((track for track in self.tracks_list
+                     if track.ipod_track.podcast_rss == episode.channel.url and
+                     track.ipod_track.podcast_url == episode.url), None)
+
     def remove_track(self, track):
         self.notify('status', _('Removing %s') % track.title)
-        self.remove_track_gpod(track.libgpodtrack)
-
-    def remove_track_gpod(self, track):
-        filename = gpod.itdb_filename_on_ipod(track)
-
+        logger.info('Removing track from iPod: %r', track.title)
+        track.ipod_track.remove_from_device()
         try:
-            gpod.itdb_playlist_remove_track(self.podcasts_playlist, track)
-        except:
-            logger.info('Track %s not in playlist', track.title)
-
-        gpod.itdb_track_unlink(track)
-        util.delete_file(filename)
+            self.tracks_list.remove(next((sync_track for sync_track in self.tracks_list
+                                          if sync_track.ipod_track == track), None))
+        except ValueError:
+            ...
```

Comment:
Is `...` an alias for pass?

Reply:
It's the [ellipsis literal](https://docs.python.org/3/library/constants.html#Ellipsis) and can be used instead of `pass` (ellipsis can do a bit more, but here it's used as a `pass` replacement).